src/chart/layout/nodes.rs

```diff
@@ -0,0 +1,816 @@
+use paste::paste;
+use std::{
+    collections::HashMap,
+    ops::{Add, Sub},
+};
+
+use stretch::number::OrElse;
+use stretch::{
+    geometry,
+    geometry::Size,
+    node::{MeasureFunc, Node, Stretch},
+    number::Number,
+    style::*,
+};
+
+/// Trait to constrain a type to a standard numeric type
+/// to prevent ambiguity in trait definitions.
+/// Idea from: https://stackoverflow.com/questions/39159226/conflicting-implementations-of-trait-in-rust
+pub trait Numeric {}
```

Comment:
I am not sure if this is necessary. Also just curious if this is needed, can we just use https://docs.rs/num/0.4.0/num/trait.Num.html ?

Reply:
That might work! I will try it.

Reply:
It doesn't work. `Numeric` is needed to disambiguate between `T: Numeric` and `S: Into<f32>` as used in `(S,S,S,S)`. Since a tuple _could_ implement `Info<f32>`, the type system gets confused. Thats why I needed `Numeric`, which allows me to restrict to an explicit number type.